scripts/tslint/preferConstRule.ts

```diff
@@ -0,0 +1,235 @@
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new PreferConstWalker(sourceFile, this.getOptions()));
+    }
+}
+
+function isBindingPattern(node: ts.Node): node is ts.BindingPattern {
+    return !!node && (node.kind === ts.SyntaxKind.ArrayBindingPattern || node.kind === ts.SyntaxKind.ObjectBindingPattern);
+}
+
+function walkUpBindingElementsAndPatterns(node: ts.Node): ts.Node {
+    while (node && (node.kind === ts.SyntaxKind.BindingElement || isBindingPattern(node))) {
+        node = node.parent;
+    }
+
+    return node;
+}
+
+function getCombinedNodeFlags(node: ts.Node): ts.NodeFlags {
+    node = walkUpBindingElementsAndPatterns(node);
+
+    let flags = node.flags;
+    if (node.kind === ts.SyntaxKind.VariableDeclaration) {
+        node = node.parent;
+    }
+
+    if (node && node.kind === ts.SyntaxKind.VariableDeclarationList) {
+        flags |= node.flags;
+        node = node.parent;
+    }
+
+    if (node && node.kind === ts.SyntaxKind.VariableStatement) {
+        flags |= node.flags;
+    }
+
+    return flags;
+}
+
+function isLet(node: ts.Node) {
+    return !!(getCombinedNodeFlags(node) & ts.NodeFlags.Let);
+}
+
+function isExported(node: ts.Node) {
+    return !!(getCombinedNodeFlags(node) & ts.NodeFlags.Export);
+}
+
+function isAssignmentOperator(token: ts.SyntaxKind): boolean {
+    return token >= ts.SyntaxKind.FirstAssignment && token <= ts.SyntaxKind.LastAssignment;
+}
+
+function isBindingLiteralExpression(node: ts.Node): node is (ts.ArrayLiteralExpression | ts.ObjectLiteralExpression) {
+    return (!!node) && (node.kind === ts.SyntaxKind.ObjectLiteralExpression || node.kind === ts.SyntaxKind.ArrayLiteralExpression);
+}
+
+interface DeclarationUsages {
+    declaration: ts.VariableDeclaration;
+    usages: number;
+}
+
+class PreferConstWalker extends Lint.RuleWalker {
+    private inScopeLetDeclarations: ts.Map<DeclarationUsages>[] = [];
+    private errors: Lint.RuleFailure[] = [];
+    private markAssignment(identifier: ts.Identifier) {
+        const name = identifier.text;
+        for (let i = this.inScopeLetDeclarations.length - 1; i >= 0; i--) {
+            const declarations = this.inScopeLetDeclarations[i];
+            if (declarations[name]) {
+                declarations[name].usages++;
+                break;
+            }
+        }
+    }
+
+    visitSourceFile(node: ts.SourceFile) {
+        super.visitSourceFile(node);
+        // Sort errors by position because tslint doesn't
+        this.errors.sort((a, b) => a.getStartPosition().getPosition() - b.getStartPosition().getPosition()).forEach(e => this.addFailure(e));
+    }
+
+    visitBinaryExpression(node: ts.BinaryExpression) {
+        if (isAssignmentOperator(node.operatorToken.kind)) {
+            this.visitLHSExpressions(node.left);
+        }
+        super.visitBinaryExpression(node);

```

Comment:
curious why you call the `super.visitBinaryExpression` after ? or it doesn't matter?